spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -57,6 +72,7 @@ public class SparkV2Filters {
   private static final String FALSE = "ALWAYS_FALSE";
   private static final String EQ = "=";
   private static final String EQ_NULL_SAFE = "<=>";
+  private static final String NOT_EQ = "<>";
```

Comment:
Is this related like (push down of <>)?  Could we do it in another pr?

Reply:
also cc @huaxingao, do you know why we didnt support it in the initial implementation?

Reply:
I think I accidentally missed this one

Reply:
I think that this was carried over from `SparkFilters`. I seem to remember not having a `NotEq` filter defined at the time so Spark would always use `Not(Eq(...))`. Looks like there is one now. I'm fine adding this here or in another PR. A separate one would be cleaner.

Reply:
Thanks @szehon-ho @huaxingao @rdblue, submit #7898 for this.